realign goals, questions, and content

tkphd opened the pull request:
Some initial work on streamlining & refactoring hpc-intro.

Closes #388.
Addresses #387.

bkmgit commented:
Link to HPC Jargon slides is nice. Some of the wording can be improved. For example:
- https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L65 should become 'familiar.' as it the level of familiarity can differ quite substantially
- https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L82 should become 'Computation at larger ...'
- https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L106 'Oss' -> 'operating systems' minimize use of abbreviations which have not been explained
- https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L123 is unclear, one can setup a homogeneous cluster in the cloud, and even if infiniband is not available, there are improved network connectivity options. VM is jargon that should be replaced by virtualization and a brief description of that this is and possibly how it might impact performance.  Apart from virtualization probably the largest other practical difference is that typically you need to have greater familiarity with software setup in the cloud than you do on an HPC system for science applications.  Cloud originally evolved from providing web services, but is now also used for big data workloads.
- https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L147 should be updated to describe an HPC cluster
Note that usually for jargon busting exercises, participants each list some terms they have come across on HPC/Supercomputing and then look up more information online and discuss it in small groups.

bkmgit commented on _episodes/11-connecting.md:
```suggestion
   character classes, 25 characters or longer. [KeePass](https://keepass.info/) and [BitWarden](https://bitwarden.com/) are two
```
Make it easier for participants to use these

bkmgit commented on _episodes/11-connecting.md:
```suggestion
* your private key (`~/.ssh/id_rsa`): *do not share with anyone!*
```

tkphd replied:
ed25519 produces a shorter key with equivalent strength to RSA; in our section immediately above, the ssh-keygen command is `ssh-keygen -a 100 -f ~/.ssh/id_ed25519 -t ed25519`, therefore the keyfile names are correct as-is.

bkmgit commented on _episodes/11-connecting.md:
```suggestion
* the shareable public key (`~/.ssh/id_rsa.pub`): if a system administrator
```

tkphd replied:
ed25519 produces a shorter key with equivalent strength to RSA; in our section immediately above, the ssh-keygen command is `ssh-keygen -a 100 -f ~/.ssh/id_ed25519 -t ed25519`, therefore the keyfile names are correct as-is.

bkmgit commented on _episodes/11-connecting.md:
This is repeated multiple times. Maybe the section can be restructured to minimize repetition?

tkphd replied:
Agreed, streamlined in 6c3971f.

bkmgit commented on _episodes/11-connecting.md:
On some systems, your ssh key will be installed for you and maybe in a different location.  This section maybe site specific, so instructors may need to adapt it to their situation.

tkphd replied:
Good point! Addressed by 90f252d.

bkmgit commented on _episodes/11-connecting.md:
```suggestion
have the Agent remember it for some number of hours or until you log off. Unless
```

bkmgit commented on index.md:
```diff
@@ -22,10 +22,11 @@ how to use them effectively.
-  use software through environment modules.
+* Use the UNIX shell (also known as terminal or command line) to
+connect to a cluster.
+* Download files from the internet directly on a cluster.
```
```suggestion
* Transfer files to a cluster.
```
Some clusters do not have direct internet access.

bkmgit commented on _episodes/13-scheduler.md:
```diff
@@ -355,36 +181,15 @@ Submit the job and monitor its status:
->
-> > ## Hint
-> >
-> > You can use the *manual pages* for {{ site.sched.name }} utilities to find
-> > more about their capabilities. On the command line, these are accessed
-> > through the `man` utility: run `man <program-name>`. You can find the same
-> > information online by searching > "man <program-name>".
```
Introducing the manual is helpful for future self directed learning, would leave some portion of trying it out as an exercise.

tkphd replied:
Agreed, but
1. `man` is covered by Shell Intro and has therefore fallen out of scope, and
2. these lines are a nested callout in an exercise that was cut in its entirety.

bkmgit commented on _episodes/14-modules.md:
```diff
@@ -6,6 +6,7 @@ questions:
+- "Understand how the module mechanism interacts with the shell."
 keypoints:
 - "Load software with `module load softwareName`."
 - "Unload software with `module purge`"
```
Note  that module purge would unload all modules, some of which may be default modules loaded upon login.  May want to explain how to unload a single module as well.

tkphd replied:
Replaced `purge` with `unload`, thanks!

bkmgit commented on _episodes/14-modules.md:
```diff
@@ -147,6 +148,18 @@ required software dependencies.
 {% include {{ site.snippets }}/modules/software-dependencies.snip %}
 
+Note that this module loading process happens principally through
+the manipulation of environment variables like `$PATH`. There
```
This maybe unclear to someone who has not used Linux/Unix very much. Perhaps one could examine the output of `echo $PATH` before and after loading a module to explain this better.

tkphd replied:
For sure! This is examined a few lines above, inside of `{{ site.snippets }}/modules/python-module-path.snip %}`.

bkmgit commented on _episodes/15-transferring-files.md:
```diff
@@ -12,15 +12,15 @@ keypoints:
-to or from the cluster. There are several options for transferring data between
-computing resources, from command line options to GUI programs, which we will
-cover here.
+Computing with a remote computer offers very limited utility if we cannot 
```
```suggestion
Using a remote computer offers very limited utility if we cannot 
```

bkmgit commented on _episodes/15-transferring-files.md:
```diff
@@ -12,15 +12,15 @@ keypoints:
+Computing with a remote computer offers very limited utility if we cannot 
+get files to or from the cluster. There are several options for 
+transferring data between computing resources, from command line 
+options to GUI programs, which we will cover here.
```
```suggestion
options to GUI programs, a few of which we will cover here.
```

bkmgit commented on _episodes/15-transferring-files.md:
```diff
@@ -200,37 +203,6 @@ A trailing slash on the target directory is optional, and has no effect for
-> > {: .solution}
-> {: .challenge}
-{: .callout}
-
```
This callout should probably remain - not all instructors will discuss it, but it on a number of clusters, specific ports need to be used to transfer files.

tkphd replied:
Cool, restored in 6c3971f.

bkmgit commented on _episodes/17-resources.md:
```diff
@@ -1,15 +1,16 @@
+exercises: 20
 questions:
-- "How do we monitor our jobs?"
-- "How can I get my jobs scheduled more easily?"
```
May want to leave this. On many systems there are different queues - even on a homogeneous system, there is usually a production and debug queue.  Perhaps ask participants to see how many queues are available and discuss their possible purposes.

tkphd replied:
We can't leave these questions in place, as the existing lesson does not answer them. Discussion of available queues is a good idea, but out of scope for this PR.

bkmgit reviewed (commented):
Thanks for this update. Have not reviewed section 16 since it will be replaced.

tkphd commented:
> Link to HPC Jargon slides is nice. Some of the wording can be improved. For example:
> 
> * https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L65 should become 'familiar.' as it the level of familiarity can differ quite substantially
> * https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L82 should become 'Computation at larger ...'
> * https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L106 'Oss' -> 'operating systems' minimize use of abbreviations which have not been explained
> * https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L123 is unclear, one can setup a homogeneous cluster in the cloud, and even if infiniband is not available, there are improved network connectivity options. VM is jargon that should be replaced by virtualization and a brief description of that this is and possibly how it might impact performance.  Apart from virtualization probably the largest other practical difference is that typically you need to have greater familiarity with software setup in the cloud than you do on an HPC system for science applications.  Cloud originally evolved from providing web services, but is now also used for big data workloads.
> * https://github.com/tkphd/hpc-intro/blob/lesson-objectives-review/files/jargon.html#L147 should be updated to describe an HPC cluster
>   Note that usually for jargon busting exercises, participants each list some terms they have come across on HPC/Supercomputing and then look up more information online and discuss it in small groups.

The jargon presentation is not exactly in-scope for this PR, but I've addressed most of these comments in 4089f66. Note that text between `???` and `---` represents author notes, which will not be rendered by default.

tobyhodges commented on _config.yml:
```diff
@@ -63,7 +63,8 @@ sched:
 
 episode_order:
-  - 11-hpc-intro
+  - 10-hpc-intro
+  - 11-connecting
```
If you are using numbered episode files, `episode_order` is redundant. This is not a problem exactly, but I wanted to clarify.

tobyhodges commented on _episodes/11-connecting.md:
```diff
@@ -0,0 +1,427 @@
+The first step in using a cluster is to establish a connection from our laptop
+to the cluster. When we are sitting at a computer (or standing, or holding it
+in our hands or on our wrists), we have come to expect a visual display with
+icons, widgets, and perhaps some windows or applications: a graphical user
+interface, or GUI. Since computer clusters are remote resources that we connect
+to over often slow or laggy interfaces (WiFi and VPNs especially), it is more
+practical to use a command-line interface, or CLI, in which commands and
+results are transmitted via text, only. Anything other than text (images, for
```
```suggestion
icons, widgets, and perhaps some windows or applications: a _graphical user
interface_, or GUI. Since computer clusters are remote resources that we connect
to over often slow or laggy interfaces (WiFi and VPNs especially), it is more
practical to use a _command-line interface_, or CLI, in which commands and
results are transmitted via text, only. Anything other than text (images, for
```

add italics to emphasis the comparison you are making

tobyhodges commented on _episodes/12-cluster.md:
```diff
@@ -1,455 +1,38 @@
+- "Compare & contrast resources on the local machine, login node, and worker
+nodes"
+- "Learn about the various filesystems on the cluster using `df`"
+- "See `who` else is logged in"
```
```suggestion
- "Find out `who` else is logged in"
```

more active than "see"

tobyhodges commented on _episodes/12-cluster.md:
```diff
@@ -1,455 +1,38 @@
+nodes"
+- "Learn about the various filesystems on the cluster using `df`"
+- "See `who` else is logged in"
+- "Glance at the number of idle and occupied nodes"
```
```suggestion
- "Assess the number of idle and occupied nodes"
```

As far as I can tell, an exploration of the output of these kinds of monitoring commands is not really written into the episode? Assuming that the Instructor will take time to explain this in a workshop setting, to help learners understand how the output should be parsed, I think "assess" represents the skill they are learning fairly well.

hot007 replied:
In the interests of security, some HPCs are starting to block a user's view of system use, e.g. `qstat`/`squeue` only show the user's own jobs. It may be necessary to tune the lesson to show whatever information/visualisation for the local system may be available. I think the intent here is perhaps along the lines of "Learn about shared system utilisation"?

tkphd replied:
Right, this episode looks at the output of `sinfo` or `pbsnodes` to see if any nodes are idle. I agree that "assess" is a good descriptor. This is not a job- or user-level inspection, so hopefully skirts the security controls?

tobyhodges commented on _episodes/13-scheduler.md:
```diff
@@ -1,23 +1,21 @@
   cluster?"
 objectives:
-- "Run a simple Hello World style program on the cluster."
 - "Submit a simple Hello World style script to the cluster."
```
```suggestion
- "Submit a simple script to the cluster."
```

tobyhodges commented on _episodes/16-parallel.md:
```diff
@@ -1,28 +1,36 @@
-- "Understand how to run a parallel job on a cluster."
+- "Construct a program that can execute in parallel."
+- "Prepare a job submission script for the parallel executable."
+- "Launch jobs with increasing degrees of parallel execution."
+- "Record and summarize the timing and accuracy of the results."
```
```suggestion
- "Launch jobs with parallel execution."
- "Record and summarize the timing and accuracy of jobs."
- "Describe the relationship between job parallelism and performance."
```

I think these are three separate skills, and should be captured in three different objectives: you are teaching learners how to 1. run a  job in parallel, 2. how to access and parse information about the performance of a job, and 3. how to translate that into an understanding of the different ways that parallel execution can effect job performance.

tobyhodges commented on _episodes/17-resources.md:
```diff
@@ -1,15 +1,16 @@
-- "Understand how to look up job statistics and profile code."
-- "Understand job size implications."
+- "Look up job statistics."
+- "Use past performance to make more accurate resource requests in job scripts."
```
```suggestion
- "Make more accurate resource requests in job scripts based on data describing past performance."
```

tkphd replied:
Wordy, but more meaningful.

tobyhodges commented on _episodes/15-transferring-files.md:
```diff
@@ -8,24 +8,23 @@ objectives:
 - "Be able to transfer files to and from a computing cluster."
```
```suggestion
- "Transfer files to and from a computing cluster."
```

tobyhodges commented on _episodes/14-modules.md:
```diff
@@ -6,9 +6,10 @@ questions:
 - "How do we load and unload software packages?"
 objectives:
 - "Understand how to load and use a software package."
+- "Understand how the module mechanism interacts with the shell."
```
```suggestion
- "Load and use a software package."
- "Explain how the module mechanism interacts with the shell."
```

I think this second one can be more specific TBH - interacts how? In what respect? In terms of affecting the PATH/environment? Or something else?

tkphd replied:
Right, the exercise in the snippets shows changes to PATH after doing a `module load`. Perhaps:
> "Explain how the shell environment changes when the module mechanism loads or unloads packages."

tobyhodges replied:
Wordy, but more meaningful 😁 

If you update the objective as you suggest, I would be happy to approve this PR

tkphd replied:
Done! b3c4a2c

tobyhodges reviewed (changes requested):
I mostly limited my review to your learning objectives.

I have made a few suggestions where I think the objectives should be reworded or - in one case - split up. A couple points to set out on my reasoning:

1. Objectives should be written as endings to a sentence that begins "By the end of this episode, learners should be able to ..."
2. The verbs at the beginning of an objective should be _active_: it is important they are directly measurable/observable because then you can design a formative assessment to evaluate learner attainment during a workshop. Verbs like "understand" and "appreciate" are not directly measurable and should be replaced. (I will submit a separate PR to take care of this in `18-responsibility`.)

ocaisa reviewed (changes requested):
I'll do my review in a few chunks

ocaisa commented on _episodes/12-cluster.md:
```diff
@@ -528,16 +109,26 @@ upgrade your OS.
+resource-intensive tasks. You should be alert to this, and check with your
+site's operators or documentation for details of what is and isn't allowed. It
+is well suited for uploading and downloading files, setting up software, and
+running tests. Generally speaking, in these lessons, we will avoid running jobs
```
```suggestion
running short tests that are not resource intensive. Generally speaking, in these lessons, we will avoid running jobs
```

tkphd replied:
This feels overly prescriptive: the criterion is not run-time or CPU/RAM usage, but rather impact on other users and the ability of the login node to do serve its primary role. As the central point of entry onto a system that may have a different architecture or software stack than a user's off-cluster resources, it is an appropriate place to build software and run test suites, some of which are time-consuming and perform a non-trivial amount of computational work, and that is right and proper.

ocaisa commented on _episodes/13-scheduler.md:
```diff
@@ -1,23 +1,21 @@
 ---
-title: "Working with the scheduler"
+title: "Queuing System Fundamentals"
```
This title change introduces a mismatch with the first question

hot007 replied:
Could it be renamed "Scheduling system fundamentals"? Keeps the gentler gist of the new title while retaining the word 'scheduler' to align with the first question.

tkphd replied:
Rephrased the title & Q1 in f141db3.

ocaisa commented on _episodes/13-scheduler.md:
```diff
@@ -1,23 +1,21 @@
-- "Everything you do should be run through the scheduler."
 - "A job is just a shell script."
-- "If in doubt, request more resources than you will need."
+- "Request slightly more resources than you will need."
```
```suggestion
- "Request _slightly_ more resources than you will need."
```

ocaisa commented on _episodes/15-transferring-files.md:
```diff
@@ -90,15 +92,14 @@ remote computer. We can leave it at that if we don't care where the file goes.
+initiated from the *outside*. This means that the `curl` command will fail, as
+an address outside the firewall is unreachable from the inside. To get around
+this, run the `curl` or `wget` command from your local machine to download the
+file, then use the `scp` command (just below here) to upload it to the cluster.
```
```suggestion
file, then use the `scp` command to upload it to the cluster.
```

ocaisa commented on _episodes/15-transferring-files.md:
```diff
@@ -90,15 +92,14 @@ remote computer. We can leave it at that if we don't care where the file goes.
 > {: .solution}
 {: .challenge}
 
+Some computer clusters are behind firewalls set to only allow transfers
```
What's a firewall?

ocaisa replied:
That doesn't mean people understand what it does, `ssh` is a thing since 1995!

ocaisa replied:
Having said that, if we introduce the image which includes the firewall that might be enough of a hint

tkphd replied:
Attempted clarification in af88aeb.

ocaisa commented on _episodes/16-parallel.md:
```diff
@@ -1,28 +1,36 @@
-&#960;, the ratio of the circumference to the diameter of a circle.
-The program generates a large number of random points on a 1&times;1 square
-centered on (&frac12;,&frac12;), and checks how many of these points fall
+π, the ratio of the circumference to the diameter of a circle.
```
Aren't these rendered characters a potential source of editor pain?

tkphd replied:
Yes, but after switching them all to HTML codes, the raw Markdown became much more difficult to understand! My suggestion for those with uncooperative editors -- and I know this sounds snarky, I mean it with tenderness & affection -- is to upgrade your terminal and/or editor, or use the Web interface.

ocaisa commented on _episodes/18-responsibility.md:
```diff
@@ -104,12 +103,12 @@ only use 100 of them! This problem can be compounded when people write scripts
-extremely frustrating! Most systems provide dedicated resources for testing
-that have short wait times to help you avoid this issue.
+Most systems provide dedicated resources for testing that have short wait times 
+to help you avoid this issue.
```
```suggestion
to help you avoid these kinds of issues.
```

ocaisa reviewed (changes requested):
Lots of effort gone in here, thanks @tkphd!

ocaisa commented on _episodes/11-connecting.md:
```diff
@@ -0,0 +1,427 @@
+
+If you have ever opened the Windows Command Prompt or macOS Terminal, you have
+seen a CLI. If you have already taken The Carpentries' courses on the UNIX
+Shell or Version Control, you have used the CLI on your local machine somewhat
```
I worked on something a bit (in a slide deck), we could introduce it piece by piece
![local_remote](https://user-images.githubusercontent.com/3840437/168813031-20d72782-9d1f-4ebb-a74c-b8739059f9b0.png)

tkphd replied:
"Local" machine/device/computer is used a number of times in the first episode. Perhaps create an issue to place this image in there?

hot007 commented on _episodes/13-scheduler.md:
```diff
@@ -37,63 +35,6 @@ why sometimes your job do not start instantly as in your laptop.
-> {: .challenge}
-{: .callout}
-
 The scheduler used in this lesson is {{ site.sched.name }}. Although
```
Perhaps here, or perhaps in the paragraph above, emphasise *scheduler* as new vocab as with other key words in the follow section.

hot007 commented:
I did not have time to do a full review, I am sorry, but I've read through the lessons and I think overall the structure and pacing is really good. I share some concerns about managing the introduction of jargon terms, but overall I think this is a much gentler approach to introducing HPC concepts than others I've seen (including lessons I've taught!!). I think all the core concepts are covered (assuming someone delivering it uses the suggested shell and python lessons as well), and it progresses nicely up to showing the user how to write their own MPI code, not just running an existing MPI programme, which is really nice. It's really good that issues of data transfer and considerate resource use are covered well.
I think overall this looks like a really great lesson sequence, I hope many HPC facilities will take it up for their own carpentries-supported training.

reid-a commented on _episodes/10-hpc-intro.md:
```diff
@@ -51,96 +52,12 @@ In all these cases, access to more (and larger) computers is needed. Those
 ## Jargon Busting Presentation
 
-Open the [HPC Jargon Buster]({{ site.baseurl }}/files/jargon.html#p1) in
+Open the [HPC Jargon Buster]({{ site.url }}{{ site.baseurl }}/files/jargon.html#p1) in
```
So what I was thinking when we did this was that some of the feedback we got from the NIST instructional sessions were that the jargon was opaque when initially presented -- people don't have anything to anchor it to, cognitively speaking. The second run with the presentation got much better feedback, the diagram helps learners build a mental model of the pieces that make up the system.

That said, for formatting reasons, the text on the slides is abbreviated, and e.g. in the first cut not all the acronyms were fully expanded. This was intentional (or at least not forbidden), the slides are meant to be the skeleton of an oral presentation, and part of the job of the presenter is to build the bridges between the slides, and describe the parts of the HPC system that are not explicit in the pictures. The slides are not the presentation, they're what you show while you give the presentation, the deeper content is in the presenter notes.

So my take-away is that in-line vocabulary in the lesson was not valuable to the learners, and that the presentation *is* valuable, but the presentation is not something that can be easily ingested off-line.

So maybe some kind of off-line version of the presentation is appropriate? But I think restoring the content we removed in favor of the presentation is probably not the right approach.

tkphd replied:
@tobyhodges suggests recording the jargon presentation and embedding it, to make it easier for self-directed learners and general Carpentries instructors get through it.

bkmgit commented:
@hot007 Thanks for the comments. Introduction to MPI programming will likely be moved to a different lesson.

tkphd commented on _episodes/11-connecting.md:
```diff
@@ -0,0 +1,427 @@
+to over often slow or laggy interfaces (WiFi and VPNs especially), it is more
+practical to use a command-line interface, or CLI, in which commands and
+results are transmitted via text, only. Anything other than text (images, for
+example) must be written to disk and opened with a separate program.
```
Behold, a rabbit hole! What's written here (and in the suggested edit) is not wrong, but the phrasing is imprecise and absolutist.
* While a vanilla SSH connection can only display text, X forwarding is built-in to most clients. Image viewers, music players, and application GUIs _can_ be opened from the terminal, and some software (_ranger_) can display images on the terminal anyway.
* Images, data files, video, etc. has to be written to memory (disk or RAM) _somewhere_, and transferred _somehow_, possibly coming through an intermediate application, before arriving intelligibly on the local monitor.

Since this whole process can be accomplished in many creative ways, I suggest revising this to soften the language and more clearly state what we're trying to say -- which is that, for the purposes of this lesson, we're only going to work with text-based data in a vanilla shell session.

tkphd replied:
Attempted to address in 1967dee.

tkphd commented on _episodes/11-connecting.md:
```diff
@@ -0,0 +1,427 @@
+If you have ever opened the Windows Command Prompt or macOS Terminal, you have
+seen a CLI. If you have already taken The Carpentries' courses on the UNIX
+Shell or Version Control, you have used the CLI on your local machine somewhat
+extensively. The only leap to be made here is to open a CLI on a *remote*
```
Yep. It is italicized for emphasis, to differentiate it from the _local_ machine we're connecting from.

tkphd commented on _episodes/11-connecting.md:
True. We can either drop "as a background process" and hope nobody wonders what the `$(...)` syntax means, or explain something about daemonizing the agent. It's not exactly backgrounded, as that implies a terminal `&`, but as part of the current shell, its output would still appear at random and often inconvenient times. We're spinning up a sub-shell for the agent. This probably requires word-smithing.

tkphd replied:
Attempted to address in 1967dee.

tkphd commented on _episodes/11-connecting.md:
True... my impulse is to hang onto it as a breadcrumb for people configuring SSH on their own machines separately, but there are myriad tutorials for that elsewhere. We can replace it with a notice to follow the documentation, send your .pub to the sysadmin, or use the web portal, possibly with a top-level template parameter to switch between the options.

tkphd commented on _episodes/11-connecting.md:
Right, the intent here is that the SSH Agent always prompts for the _key_ password the first time it's loaded, but not thereafter. We should clarify, but retain this, because re-prompts indicate an Agent failure.

ocaisa commented on _episodes/11-connecting.md:
```diff
@@ -0,0 +1,427 @@
+set these arguments before clicking "connect." From the terminal, you'll write
+something like `ssh userName@hostname`, where the argument is just like an
+email address: the "@" symbol is used to separate the personal ID from the
+address of the shared resource.
```
@tkphd I think this one is still valid

ocaisa commented on _episodes/11-connecting.md:
Also still valid I would say

ocaisa commented on _episodes/11-connecting.md:
Also still valid I would say

ocaisa commented on _episodes/11-connecting.md:
Also still valid

ocaisa commented on _episodes/11-connecting.md:
Still valid

ocaisa commented on _episodes/13-scheduler.md:
```diff
@@ -446,12 +250,11 @@ walltime, and attempt to run a job for two minutes.
+{{ site.sched.comment }} {{ site.sched.flag.time }} 00:01 # timeout in HH:MM
 
 echo "This script is running on ... "
 sleep 120 # time in seconds
```
```suggestion
sleep 240 # time in seconds
```

ocaisa reviewed (approved):
LGTM

tkphd commented:
Closing to try to trigger the checks to actually run. Bear with me ...
